src/main/java/com/google/sps/data/OrganizationUpdater.java

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Arrays;
+import com.google.appengine.api.datastore.Entity;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+
+public final class OrganizationUpdater {
+
+  private Entity entity;
+
+  public OrganizationUpdater(Entity entity) {
+     this.entity = entity;
+  }
+
+  public Entity getEntity() {
+      return this.entity;
+  }
+
+  public void updateOrganization(HttpServletRequest request, boolean isMaintainer, boolean isModerator) throws IllegalArgumentException{
+    Set<String> requiresMaintainer = new HashSet<String>();
+    Set<String> requiresModerator = new HashSet<String>();
+    Map<String, String> properties = new HashMap<String,String>();
+
+    requiresMaintainer.add("isApproved");
+    requiresModerator.add("moderatorList");
+
+    // Format is Form Name, Entity Property 
+    properties.put("org-name", "orgName");
+    properties.put("org-email", "orgEmail");
+    properties.put("org-street-address", "orgStreetAddress");
+    properties.put("org-phone-num", "orgPhoneNum");
+    properties.put("org-url", "orgUrl");
+    properties.put("org-description","orgDescription");
+    properties.put("approval", "isApproved");
+    properties.put("moderator-list", "moderatorList");
+
+    for(Map.Entry<String, String> entry : properties.entrySet()) {
+      String formKey = entry.getKey();
+      String propertyKey = entry.getValue();
+      String formValue = "";
+
+      // will only get approval param is a maintainer sent the request
+      if(requiresMaintainer.contains(propertyKey) && !isMaintainer) {
+        continue;
+      }
+
+      // will only get moderatorList param if request is sent by a moderator or maintainer
```

Comment:
"or" here doesn't line up with this if statement

Reply:
I think it does because it would do the continue if a request was not sent by a moderator or a maintainer. The check would fail if it was a moderator or then fail if it was a maintainer. I think the key here is that the isModerator check is before the isMaintainer check.

Reply:
OK, I think you're right. Is there a way to rewrite this so there are less "nots"?
Eg currently this is
allow access = not(a && not(b) && not(c))

Reply:
yes i think its currently a && !b && !c so it would be  a && not( b || c) by demorgans law